nncf/experimental/onnx/statistics/aggregator.py

```diff
@@ -45,12 +43,10 @@ def collect_statistics(self, model: onnx.ModelProto) -> None:
         for _, v in self.layers_statistics.items():
             max_number_samples = max(max_number_samples, v.num_samples)
 
-        with tempfile.NamedTemporaryFile() as temporary_model:
-            onnx.save(model_with_intermediate_outputs, temporary_model.name)
-            self.engine.set_model(temporary_model.name)
-            self.engine.sampler = create_onnx_sampler(self.dataloader, range(max_number_samples))
-            output = self.engine.compute_statistics(self.layers_statistics)
-            self._agregate_statistics(output, self.layers_statistics)
+        self.engine.set_model(model_with_intermediate_outputs)
+        self.engine.sampler = create_onnx_sampler(self.dataloader, range(max_number_samples))
```

Comment:
Why should you set Sampler for Engine when it is created inside `compute_statistics()`?

Reply:
This moment is need to be updated. Currently, it's WA and not a general solution, for the case when the sampler wasn't set before calculation. I see two possible ways to improve that: create `sampler` validation inside `compute_statistics()` and raise an exception or remove `sampler` creation outside the Engine methods. 
But currently, there is no way to get max_number_samples internally in the Engine. This is why `create_onnx_sampler()` calls twice here.